src/app/machines/CreateConnectorWizard.machine.ts

```diff
@@ -83,10 +102,73 @@ const model = createModel({} as Context, {
   },
 });
 
+// Guards
+const isConnectorSelected: WizardGuard = (context, event) => {
+  const subStep = (event as { subStep?: number }).subStep;
+  if (subStep) {
+    return (
+      context.selectedConnector !== undefined &&
+      (context.connectorConfiguration !== undefined ||
+        subStep <= context.activeConfigurationStep!)
+    );
+  }
+  return context.selectedConnector !== undefined;
+};
+
+const isKafkaInstanceSelected: WizardGuard = (context, _event) =>
+  context.selectedKafkaInstance !== undefined;
+
+const isNamespaceSelected: WizardGuard = (context, _event) =>
+  context.selectedNamespace !== undefined;
+
+const isConnectorConfigured: WizardGuard = (context, _event) => {
+  if (!context.configurationSteps) {
+    return (
+      context.connectorConfiguration !== undefined &&
+      context.connectorConfiguration !== false
+    );
+  }
+  return (
+    (context.connectorConfiguration !== undefined &&
+      context.connectorConfiguration !== false) ||
+    (context.activeConfigurationStep ===
+      context.configurationSteps.length - 1 &&
+      context.isConfigurationValid === true)
+  );
+};
+
+const isCoreConfigurationConfigured: WizardGuard = (context, _event) => {
+  return (
+    context.name !== undefined &&
+    context.name.length > 0 &&
+    context.userServiceAccount !== undefined &&
+    context.userServiceAccount.clientId.length > 0 &&
+    context.userServiceAccount.clientSecret.length > 0
+  );
+};
+
+const isErrorHandlerConfigured: WizardGuard = (context, _event) =>
+  context.userErrorHandler !== undefined &&
+  context.userErrorHandler === 'dead_letter_queue'
+    ? context.topic !== undefined && context.topic.length > 0
+    : (context.topic !== undefined && context.topic.length > 0) ||
+      context.userErrorHandler !== undefined;
+
+const canConfigurationBeSaved: WizardGuard = (context, event) => (
+  isConnectorSelected(context, event),
+  isKafkaInstanceSelected(context, event),
+  isNamespaceSelected(context, event),
+  isConnectorConfigured(context, event),
+  isCoreConfigurationConfigured(context, event),
+  isErrorHandlerConfigured(context, event)
```

Comment:
Just checked the flow for a debezium connector and as I was suspecting there's some adjustments to make still.

@indraraj this probably isn't a valid check for debezium is it?  I figure at this point I can check the selected connector type within here and not make this check.  Anything else I should be aware of?

Reply:
I gotta update this anyways, it's not right :-)

Reply:
Ok, fixed that.